pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordenricher/RecordEnricher.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.recordenricher;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+public abstract class RecordEnricher {
```

Comment:
should we put this in segment-spi? do we plan to have other type of enricher plugin in the future?

Reply:
I'm not sure if we'd have support enricher as plugins, I don't see transformers plugin being supported. But yes we may have more enrichers in the future.

Reply:
What's the reason for not supporting transformer plugins?